storage/storage.go

```diff
@@ -686,3 +697,10 @@ func WithPrefixInBlobPath(prefix string) LeaserCheckpointerOption {
 		return nil
 	}
 }
+
+func WithInitialCheckpoint(getInitialCheckpoint func() persist.Checkpoint) LeaserCheckpointerOption {
```

Comment:
Please add a comment explaining what this option is for

Reply:
Updated, rebased and squashed 👍